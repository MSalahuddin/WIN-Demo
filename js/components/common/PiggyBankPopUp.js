import React, { useState } from 'react';
import Modal from 'react-native-modal';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import * as RNIap from 'react-native-iap';
import LinearGradient from 'react-native-linear-gradient';
import { SafeAreaView, Platform } from 'react-native';
import Text, { SIZE, FONT_FAMILY } from './Text';
import Button, { BUTTON_COLOR_SCHEME } from './Button';
import IconButton from './IconButton';
import { scaleHeight, scale, scaleWidth,heightRatio } from '../../platformUtils';
import { color } from '../../styles';
import { popUpStrings } from '../../stringConstants';
import { getPercentage, formatTokensLabel } from '../../utils';
import {
  ExitCircle,
  empty,
  full,
  piggyBankHeader,
  inviteFriendBackgroundTokenImage1,
  inviteFriendBackgroundTokenImage2,
  upRollToken,
  downToken,
  backGroundPiggyBank
} from '../../../assets/images';

const Background = styled.ImageBackground`

  align-items: center;
  background-color: ${color.white};
  border-bottom-left-radius: ${scaleHeight(8)};
  border-bottom-right-radius: ${scaleHeight(8)};
  justify-content: flex-start;
  margin-top: ${-scaleHeight(20)};
  padding-top: ${Platform.OS==='android'?scaleHeight(10):scaleHeight(25)};
  padding-bottom: ${scaleHeight(65)};
  border-width: 1;
  border-color: ${color.white};
  border-radius: 15;
  shadow-color: ${color.blackShadow};
  shadow-offset: 4px 4px;
  shadow-opacity: 1;
  shadow-radius: 2px;
  width: 85%;
  z-index: -1;
`;

const ExitCircleBtn =Platform.OS==='android'? styled(IconButton)`
z-index: 1;
justify-content:flex-end;
margin-right:${scaleWidth(-283)};
`:
styled(IconButton)`
z-index: 1;
justify-content:flex-end;
margin-right:${scaleWidth(-290)}
`
;
const ButtonContentContainer = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: center;
`;
const ButtonContainer = styled.View`
`;
const ButtonWrapper = styled(Button)`
  margin-top: ${({ marginTop }) => scaleHeight(marginTop)};
`;



const ButtonText = styled(Text)`
  margin-top: ${Platform.OS === 'android' ? scaleHeight(0) : scaleHeight(7)};
`;

const PiggyBankIcon =Platform.OS==='android'? styled.Image`
  margin-top:${scaleHeight(-80)}
  margin-bottom: ${scaleHeight(5)};
  margin-left:${scaleWidth(-15)};
`:
styled.Image`
margin-top:${scaleHeight(-80)}
margin-bottom: ${scaleHeight(5)};
margin-left:${scaleWidth(-15)};
`;
;
const PiggyBankTokenContainer = Platform.OS==='android'? styled.View`
margin-top:${scaleHeight(-70)}
flex-direction:row;
margin-bottom:${scale(10)}
`:
styled.View`
margin-top:${scaleHeight(heightRatio>1?-50:-60)}
flex-direction:row;
margin-bottom:${scaleHeight(10)}
`;


const ContentView = styled.View`
flex-direction:column;

`;
const PiggyBankHeadingCoinLeft = styled.Image`
  height: ${scaleHeight(50)};
  margin-top:${scaleHeight(-10)};
  position:absolute;
  left:${scaleWidth(Platform.OS==='android'?-90:-80)}
  
`;
const PiggyBankHeadingCoinRight =Platform.OS==='android'? styled.Image`
  height: ${scaleHeight(140)};
  width: ${scaleHeight(140)};
  margin-top:${scaleHeight(-30)};
  position:absolute;
  right:${scaleWidth(heightRatio>1?-105:-95)}
  
`:
styled.Image`
  height: ${scaleHeight(heightRatio > 1 ? 120: 145)};
  width: ${scaleHeight(heightRatio > 1 ? 120: 145)};
  margin-top:${scaleHeight(heightRatio>1?-25:-45)};
  position:absolute;
  right:${scaleWidth(heightRatio>1?-105:-100)}
  
`;
;
const PiggyBankProgressCoinRight = styled.Image`
height: ${scaleHeight(50)};
width: ${scaleHeight(50)};
position:absolute;
top:${scaleHeight(-15)}
right:${scaleWidth(3)}

`;
const PiggyBankProgressCoinLeft = styled.Image`
height: ${scaleHeight(105)};
width: ${scaleHeight(105)};
position:absolute;
top:${scaleHeight(30)}
left:${scaleWidth(-52)}

`;
const PiggyBankbuttonCoinright = styled.Image`
height: ${scaleHeight(40)};
width: ${scaleHeight(40)};
position:absolute;
top:${scaleHeight(10)}
right:${scaleWidth(-40)}
`;

const PopUpContentView = styled.View`
  align-items: center;
  justify-content: center;
  margin-top:${scale(Platform.OS==='android'?20:10)}
`;

const StyledModal = styled(Modal)`
  align-items: center;
  background-color: ${color.popupBlack};
  justify-content:center;
  margin: 0;
`;

const ProgressBarContainer = styled.View`
  align-items: center;
  width: 100%;
  padding-top: ${scaleHeight(10)};
  padding-bottom: ${scaleHeight(5)};
  border-radius: ${scaleHeight(8)};
`;

const ProgressBarWrapper = styled.View`
  width: 100%;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  margin-bottom:${scale(10)};
  padding-bottom: ${Platform.OS === 'android' ? scaleHeight(5) : scaleHeight(8)};

`;

const ProgressBar = styled.View`
  height: 30;
  margin-horizontal: ${scale(-7)};
  border-color:${color.borderBlue};
  width: 45%;
  border-width:4;
  z-index:2;
  shadow-opacity:1.0;
`;

const Progress = styled(LinearGradient)`
  align-items: center;
  justify-content: center;
  border-top-right-radius: ${scale(25)};
  border-bottom-right-radius: ${scale(25)};
  height: 22;
  z-index:1;
  position: absolute;
  width: ${({ widthPercentage }) => widthPercentage === 100 ? widthPercentage + 10 : widthPercentage}%;
`;

const ProgressBarWrapperImages = styled.Image`
  height: ${scaleHeight(40)};
  width: ${scaleHeight(40)};
  z-index:5
`;

const TextWrapper = styled(Text)`
  margin-horizontal: ${scale(28)};
`;
const TokenLeftTextWrapper = styled(Text)`
  text-align:center;
  margin-left:${scaleWidth(4)}
`;
const FooterTextWrapper = styled(Text)`
margin-horizontal: ${scale(28)};
margin-vertical:${scale(5)}
`;
const HeadingTextWrapper = styled(Text)`
  margin-top: ${scaleHeight(0)};
  margin-bottom: ${scaleHeight(heightRatio>1? 10: 5)};
`;
const PiggyBankTokenView = styled.View`
background-color:${color.tagBlue};
justify-content:center;
align-items:center;
width:${scaleWidth(200)};
border-radius:${scale(45)};
`;
const PiggybankTokens = styled(Text)`
  margin-top: ${scale(Platform.OS==='android'?10:20)};
  margin-bottom: ${scaleHeight(10)};
 
`;




const PiggyBankPopUp = ({ data, isVisible, onClose, notApplicable }) => {
  const calculatedCurrentToken = data?.playerPiggyBankTokens >= data?.maxTokens ? data?.maxTokens : data?.playerPiggyBankTokens
  const widthPercentage = data && getPercentage(calculatedCurrentToken, data.maxTokens);
  const requestPurchase = async productId => {
    try {
      const payload = await RNIap.getProducts([productId]);
      if (payload.length !== 0) {
        await RNIap.initConnection()
        await RNIap.requestPurchase(productId, false);
        onClose()
      }
    } catch (error) {
      // Error handled in purchase listener
    }
  };
  const getID = async () => {
    const productId = Platform.OS === 'android' ? data.androidProductId : data.iosProductId
    if (data.playerPiggyBankTokens >= data.minTokens) {
      await requestPurchase(productId)
    }
    else {
      notApplicable()
    }
  }

  return (
    <SafeAreaView>
      <StyledModal isVisible={isVisible}>

        <ExitCircleBtn
          testID="exit-circle"
          onPress={onClose}
          icon={ExitCircle}
          size={30}

        />
        <Background source={backGroundPiggyBank} resizeMode="stretch">
          <PiggyBankIcon source={piggyBankHeader} resizeMode="contain" />

          <PiggyBankTokenContainer>
            <PiggyBankHeadingCoinLeft source={inviteFriendBackgroundTokenImage1} resizeMode="contain" />
            <ContentView>
              <HeadingTextWrapper fontFamily={FONT_FAMILY.SEMI_BOLD}
                size={SIZE.LARGE} color={color.white} alignCenter>
                {popUpStrings.piggyBankYouHave}
              </HeadingTextWrapper>
              <PiggyBankTokenView>
                <PiggybankTokens fontFamily={FONT_FAMILY.SEMI_BOLD}
                  size={SIZE.NORMAL} color={color.gold} alignCenter>
                  {data && popUpStrings.HeadingOnePiggyBankAddToken(formatTokensLabel(data.playerPiggyBankTokens))}
                </PiggybankTokens>
              </PiggyBankTokenView>

            </ContentView>

            <PiggyBankHeadingCoinRight source={inviteFriendBackgroundTokenImage2} resizeMode="contain" />
          </PiggyBankTokenContainer>
         {(data && data?.playerPiggyBankTokens - data?.piggyBankValueComparisonAmount>0) && <TextWrapper fontFamily={FONT_FAMILY.BOLD} size={heightRatio > 1 ? SIZE.LARGE:SIZE.SMALL}
            color={color.white} alignCenter>
            {popUpStrings.PiggyBankMoreToken(data?.playerPiggyBankTokens - data?.piggyBankValueComparisonAmount)}
          </TextWrapper>
          }
          <PopUpContentView>

            <TextWrapper fontFamily={FONT_FAMILY.REGULAR} size={SIZE.XXSMALL}
              color={color.white} alignCenter>
              {popUpStrings.lineOnePiggyBankAddToken}
            </TextWrapper>
            <TextWrapper fontFamily={FONT_FAMILY.REGULAR} size={SIZE.XXSMALL}
              color={color.white} alignCenter>
              {popUpStrings.lineTwoPiggyBankAddToken}
            </TextWrapper>
            <TextWrapper fontFamily={FONT_FAMILY.REGULAR} size={SIZE.XXSMALL}
              color={color.white} alignCenter>
              {popUpStrings.lineSixPiggyBankAddToken}
            </TextWrapper>
            <TextWrapper fontFamily={FONT_FAMILY.REGULAR} size={SIZE.XXSMALL}
              color={color.white} alignCenter>
              {popUpStrings.lineSevenPiggyBankAddToken}
            </TextWrapper>
          </PopUpContentView>
          <ProgressBarContainer>
            <ProgressBarWrapper>
              <ProgressBarWrapperImages source={empty} />
              <ProgressBar>
                <Progress colors={[color.goldentGradientLight, color.goldentGradientMid, color.goldentGradientDark]} widthPercentage={widthPercentage} />
              </ProgressBar>
              <ProgressBarWrapperImages source={full} />
            </ProgressBarWrapper>
            <TokenLeftTextWrapper fontFamily={FONT_FAMILY.BOLD} size={SIZE.XSMALL} color={color.white} alignCenter>
              {data &&
                popUpStrings.lineFivePiggyBankAddToken((parseInt(data.maxTokens, 10) - parseInt(data.playerPiggyBankTokens, 10)), formatTokensLabel(data.maxTokens - data.playerPiggyBankTokens))}
            </TokenLeftTextWrapper>
            <PiggyBankProgressCoinRight source={inviteFriendBackgroundTokenImage2} resizeMode="contain" />
            <PiggyBankProgressCoinLeft source={upRollToken} resizeMode="contain" />
          </ProgressBarContainer>
          <ButtonContainer>
            <ButtonWrapper
              borderRadius={44}
              height={55}
              width={220}
              theme={BUTTON_COLOR_SCHEME.GREEN}
              onPress={() => getID()}
              marginTop={0}
            >
              <ButtonContentContainer>
                <ButtonText color={color.white} size={SIZE.LARGE}>
                  {data && popUpStrings.Unlock(data.localizedPrice)}
                </ButtonText>
              </ButtonContentContainer>
            </ButtonWrapper>
            <PiggyBankbuttonCoinright source={downToken} resizeMode="contain" />
          </ButtonContainer>
          <FooterTextWrapper fontFamily={FONT_FAMILY.BOLD} size={SIZE.XXSMALL} color={color.silverWhite} alignCenter>
              {(data && data?.playerPiggyBankTokens - data?.piggyBankValueComparisonAmount>0) &&
                popUpStrings.comparedText(data?.piggyBankValueComparisonTokenPack.toLowerCase())}
            </FooterTextWrapper>
        </Background>
      </StyledModal>
    </SafeAreaView>
  );
};

PiggyBankPopUp.propTypes = {
  data: PropTypes.node.isRequired,
  isVisible: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  notApplicable: PropTypes.func.isRequired
};

export default PiggyBankPopUp;
