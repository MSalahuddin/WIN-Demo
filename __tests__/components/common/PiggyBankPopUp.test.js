import React from 'react';
import { render } from 'react-native-testing-library';
import MockProvider from '../../MockProvider';
import PiggyBankPopUp from '../../../js/components/common/PiggyBankPopUp';

describe('Component PiggyBankPopUp', () => {
  const onClose = jest.fn();
  const data = {
    androidProductId: 'ww_beta_piggy_bank_golden_pack',
    iosProductId: 'ww_beta_piggy_bank_golden_pack',
    isActive: true,
    isPiggyBankUnlocked: true,
    maxTokens: 750,
    minTokens: 1,
    piggyBankId: 1,
    playerPiggyBankTokens: 0,
    tokenPercent: 2
  };

  test('show PiggyBankPopUp', () => {
    const output = render(
      <MockProvider>
        <PiggyBankPopUp data={data} isVisible onClose={onClose} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });
});
