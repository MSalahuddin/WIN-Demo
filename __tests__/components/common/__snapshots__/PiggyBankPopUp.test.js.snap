// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PiggyBankPopUp show PiggyBankPopUp 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(34,34,34, 0.6)",
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Array [
            Object {
              "height": 220,
              "justifyContent": "flex-end",
              "marginBottom": 60,
              "marginTop": 100,
              "width": 460,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        width={740}
      >
        <Image
          bannerType="COIN"
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 141.56521739130434,
                "width": 740,
              },
            ]
          }
          width={740}
        />
        <Text
          alignCenter={false}
          allowFontScaling={false}
          bannerType="COIN"
          color="#000000"
          fontFamily="BOLD"
          fontSize={22}
          isUnderlined={false}
          isUppercase={true}
          maxFontSizeMultiplier={1.3}
          size="BANNER_LARGE"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 34.5,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
                "letterSpacing": 1.5,
                "position": "absolute",
              },
            ]
          }
        >
          Piggy Bank
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomLeftRadius": 16,
              "borderBottomRightRadius": 16,
              "justifyContent": "flex-start",
              "marginTop": -40,
              "paddingBottom": 300,
              "paddingTop": 20,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "80%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="SEMI BOLD"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="NORMAL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "Calibre-Semibold",
                  "fontSize": 30,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginBottom": 20,
                  "marginTop": 40,
                },
              ]
            }
          >
            You have 0 tokens!
          </Text>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="REGULAR"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "CalibreRegular",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginHorizontal": 42,
                },
              ]
            }
          >
            Unlock your piggy bank tokens at
          </Text>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="REGULAR"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "CalibreRegular",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginHorizontal": 42,
                },
              ]
            }
          >
            anytime. Tokens are added every
          </Text>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="REGULAR"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "CalibreRegular",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginHorizontal": 42,
                },
              ]
            }
          >
            time you play. The more you play
          </Text>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="REGULAR"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "CalibreRegular",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginHorizontal": 42,
                },
              ]
            }
          >
            the better the savings!
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#7C89D8",
                "borderRadius": 16,
                "marginTop": 282,
                "paddingBottom": 100,
                "paddingTop": 60,
                "position": "absolute",
                "width": "92%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 16,
                  "width": "100%",
                },
              ]
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 50,
                    "width": 50,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#E5E5E5",
                    "borderRadius": 4.5,
                    "height": 10,
                    "marginHorizontal": 12,
                    "width": "40%",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#FFDB4F",
                      "borderRadius": 4.5,
                      "height": 10,
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": "0%",
                    },
                  ]
                }
                widthPercentage={0}
              />
            </View>
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 50,
                    "width": 50,
                  },
                ]
              }
            />
          </View>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="REGULAR"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "CalibreRegular",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginHorizontal": 42,
                },
              ]
            }
          >
            Only 750 tokens away from full!
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginVertical": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(96,202,142,1)",
              "borderRadius": 88,
              "height": 176,
              "justifyContent": "center",
              "marginTop": -92,
              "opacity": 1,
              "width": 542,
            }
          }
          testID="button"
        >
          <View
            borderRadius={44}
            colorScheme="rgba(128,229,173,1)"
            height={88}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(128,229,173,1)",
                  "borderRadius": 88,
                  "height": 166,
                  "justifyContent": "center",
                  "width": 536,
                },
              ]
            }
            width={271}
          >
            <View
              borderRadius={44}
              colorScheme="rgba(114,213,158,1)"
              height={88}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(114,213,158,1)",
                    "borderRadius": 108,
                    "height": 162,
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": 536,
                  },
                ]
              }
              width={271}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="MEDIUM"
                  isUnderlined={false}
                  isUppercase={false}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Calibre-Medium",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginTop": 14,
                      },
                    ]
                  }
                >
                  Unlock for undefined
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="button"
      >
        <Text
          alignCenter={false}
          allowFontScaling={true}
          color="#000000"
          fontFamily="MEDIUM"
          isUnderlined={true}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="XXSMALL"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Calibre-Medium",
                "fontSize": 21,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "underline",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
              Object {
                "color": "#FFFFFF",
                "marginTop": 20,
                "textDecorationColor": "#FFFFFF",
              },
            ]
          }
        />
      </View>
    </View>
  </Modal>
</RCTSafeAreaView>
`;
